### PR TITLE
Allow single quotes for attributes

### PR DESCRIPTION
**Pull request by Wiseqube**
In its current implementation the template-rewriter empties xml(html)-attributes which are quoted using a single quote, although they are valid XML. 
For example:
`<input type='text'/>` will be converted to `<input type/>`

This PR adds the ability for the template rewrite to also detect these types of attributes and parse them correctly.

### CODE REVIEW

File: src/gocept/template_rewrite/pagetemplates.py [hunk 1]
```diff
@@ -140,13 +141,26 @@ def handle_starttag(self, tag, attrs, is_short_tag=False):
...
-                # occurrence of it.
+                # check if this is an attribute with single quotes
                 mo = re.search(
-                    leading_whitespace.format(attr) + r'(\s*)', full_tag,
-                    flags=re.IGNORECASE)
-                raw_value = None
+                    leading_whitespace.format(attr) + "(?==')", full_tag,
+                    flags=re.IGNORECASE
+                )
+                if mo:
```
Comment (viktordick) [inline src/gocept/template_rewrite/pagetemplates.py:149]: I'm not sure about this control flow. You introduce some additional layers here, which prompts me to question if we really need to search first for the part with `leading_wihespace` and then again for the actually interesting pattern.
How about something like
```
mo = re.search(value_pattern...)
if not mo:
    mo = re.search(value_pattern_single...)
if not mo:
    mo = re.search(value_pattern_singleton...)
raw_value = None
if mo and mo.groups():
    raw_value = mo.group(1)
if not mo:
    parse_error = True
    break
```

Comment (viktordick) [reply src/gocept/template_rewrite/pagetemplates.py:149]: I tried it a bit and my suggestion can be seen [here](https://github.com/gocept/gocept.template_rewrite/compare/rewrite-search-cascade?expand=1) - what do you think?

Comment (Wiseqube) [reply src/gocept/template_rewrite/pagetemplates.py:149]: I like that approach, could you push these changes to my branch! To be extra fancy you could the replace the
`if not mo:
     parse_error = True
     break
` with an `else` path of the for-loop above!

Comment (viktordick) [reply src/gocept/template_rewrite/pagetemplates.py:141]: Done, although I must confess that I am not the hugest fan of Python's `for .. else`. Yes, this is the archetypal use case for it, but if someone comes from another language and does not know this feature, it seems like an error or like it means something else. So I always add an explaining comment when using that feature.

Anyhow: LGTM. Maybe test it out on the project where this whole thing became relevant before merging.

**Review by viktordick — APPROVED** [review]

**Comment by icemac** [comment]
Thank you for (reviewing) this PR. 😃

**Comment by icemac** [comment]
Just released as https://pypi.org/project/gocept.template-rewrite/1.1/

**Comment by viktordick** [comment]
Thank you @icemac !

**Pull request merged**